packages/eslint-plugin/src/rules/no-misused-spread.ts

```diff
@@ -104,6 +113,77 @@ export default createRule<Options, MessageIds>({
       }
     }
 
+    function getMapSpreadSuggestions(
+      node: TSESTree.JSXSpreadAttribute | TSESTree.SpreadElement,
+      type: ts.Type,
+    ): TSESLint.ReportSuggestionArray<MessageIds> | null {
+      const types = tsutils.unionTypeParts(type);
+      if (types.some(t => !isMap(services.program, t))) {
+        return null;
+      }
+
+      if (
+        node.parent.type === AST_NODE_TYPES.ObjectExpression &&
+        node.parent.properties.length === 1
+      ) {
+        return [
+          {
+            messageId: 'replaceMapSpreadInObject',
+            fix: getWrappingFixer({
+              node: node.parent,
+              innerNode: node.argument,
+              sourceCode: context.sourceCode,
+              wrap: code => `Object.fromEntries(${code})`,
+            }),
+          },
+        ];
+      }
+
+      return [
+        {
+          messageId: 'replaceMapSpreadInObject',
+          fix: getWrappingFixer({
+            node: node.argument,
+            sourceCode: context.sourceCode,
+            wrap: code => `Object.fromEntries(${code})`,
+          }),
+        },
+      ];
+    }
+
+    function isHigherPrecedenceThanAwait(tsNode: ts.Node): boolean {
+      const operator = ts.isBinaryExpression(tsNode)
+        ? tsNode.operatorToken.kind
+        : ts.SyntaxKind.Unknown;
+      const nodePrecedence = getOperatorPrecedence(tsNode.kind, operator);
+      const awaitPrecedence = getOperatorPrecedence(
+        ts.SyntaxKind.AwaitExpression,
+        ts.SyntaxKind.Unknown,
+      );
+      return nodePrecedence > awaitPrecedence;
+    }
```

Comment:
[Refactor] This is directly copied from `return-await`, right? Since it's a nontrivial function I think it'd be best to put in a util file and re-use between the two rules.